storagenode/piecestore/verification_test.go

```diff
@@ -0,0 +1,130 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information
+
+package piecestore_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"storj.io/storj/internal/testcontext"
+	"storj.io/storj/internal/testplanet"
+	"storj.io/storj/internal/teststorj"
+	"storj.io/storj/pkg/auth/signing"
+	"storj.io/storj/pkg/identity"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/storj"
+)
+
+func TestOrderLimitValidation(t *testing.T) {
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	planet, err := testplanet.New(t, 1, 1, 1)
+	require.NoError(t, err)
+	defer ctx.Check(planet.Shutdown)
+
+	planet.Start(ctx)
+
+	for _, tt := range []struct {
+		satelliteID     *identity.FullIdentity
+		uplinkID        storj.NodeID
+		storageNodeID   storj.NodeID
+		pieceID         storj.PieceID
+		action          pb.PieceAction
+		serialNumber    storj.SerialNumber
+		pieceExpiration time.Duration
+		orderExpiration time.Duration
+		limit           int64
+		err             string
+	}{
+		{ // unapproved satellite id
+			satelliteID:     testplanet.MustPregeneratedIdentity(0),
```

Comment:
Use `storj.NodeID{100}` or similar. Since all of them end up using the same pregenerated identities, accidentally it might still end up in the approved list.

Reply:
you mean use pregenerated identity with index 100 or create custom identity?

Reply:
Ah, it's the full identity not the ID, make the field name `satellite`, to be clearer. But yeah, in that case you can use `planet.NewIdentity()`, or if there are multiple places where you want one, then you could make `planet.MustNewIdentity()`.